BillTrack.Persistence/Repositories/DepartmentRepository.cs

```diff
@@ -0,0 +1,38 @@
+using BillTrack.Domain.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace BillTrack.Persistence.Repositories;
+
+public class DepartmentRepository : ICrudRepository<Department>
```

Comment:
it is a strictly typed repository. It is not a generic one that we talked about.

Please, replace all the typed repositories with a single generic one.
Here is an example of `Repository<T>` (named `GenericRepository<TEntity>`):
https://codingblast.com/entity-framework-core-generic-repository/

Reply:
fixed